cli/link.go

```diff
@@ -84,17 +86,47 @@ func (p PipelineLink) IsLocal() bool {
 	return p.FsAllow
 }
 
-//checks if the pipeline is up
-//otherwise it brings it up and fills the
-//link object
+// checks if the pipeline is up
+// otherwise it brings it up and fills the
+// link object
 func bringUp(pLink *PipelineLink) error {
 	alive, err := pLink.pipeline.Alive()
 	if err != nil {
-		if pLink.config[STARTING].(bool) {
-			alive, err = NewPipelineLauncher(pLink.pipeline,
-				pLink.config.ExecPath(), pLink.config[TIMEOUT].(int)).Launch(os.Stdout)
-			if err != nil {
-				return fmt.Errorf("Error bringing the pipeline2 up %v", err.Error())
+		// Requested behavior :
+		// dp2 launches pipeline-ui (if needed); dp2 calls pipeline-ui as cli handler; pipeline-ui calls dp2 with the port info as an argument
+
+		// When port attribute is set, the cli tool should not try to start a pipeline instance
+		// and directly connect to pipeline instance for which the port is set on the command line
+		if pLink.config[STARTING].(bool) && !(slices.Contains(os.Args[1:], "--port") || slices.Contains(os.Args[1:], "--host")) {
+			execpath := pLink.config.ExecPath()
+			// if execpath ends with DAISY Pipeline or DAISY Pipeline.exe
+			// we forward the command to the electron app
+			if strings.HasSuffix(execpath, "DAISY Pipeline") || strings.HasSuffix(execpath, "DAISY Pipeline.exe") {
+				args := os.Args[1:]
+				if len(os.Args) == 1 {
+					args = append(args, "help")
+				}
+				absPath, err := exec.LookPath(execpath)
+				if err != nil {
+					absPath = execpath
+				}
+				cmd := exec.Command(absPath, args...)
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				if err := cmd.Run(); err != nil {
+					return fmt.Errorf("Could not use the pipeline app: %v", err)
+				} else {
+					os.Exit(cmd.ProcessState.ExitCode())
+				}
+			} else {
+				alive, err = NewPipelineLauncher(
+					pLink.pipeline,
+					execpath,
+					10,
+				).Launch(os.Stdout)
+				if err != nil {
+					return fmt.Errorf("Error bringing the pipeline2 up %v", err.Error())
+				}
 			}
```

Comment:
I didn't think about retaining the support for other web service launcher scripts. I guess it is indeed a possibility. I'm not totally convinced of the usefulness of this feature, now that the web service is launched through the app, but it doesn't hurt either. Two comments:

- If the executable is not the app, it should be allow to have `port`/`host`/`ws_path` in the config/command.
- I'm not sure about reusing the `app_path` setting. OTOH having two settings (`app_path` and `exec_line`) is also a bit confusing.